src/scrape_config_builder.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper class to build scrape configurations for Blackbox Exporter."""
+
+from typing import Any, Dict, List
+from urllib.parse import urlparse
+
+import yaml
+
+
+class ScrapeConfigBuilder:
+    """Helper class to build scrape configurations for Blackbox Exporter."""
+
+    def __init__(self, external_url: str):
+        """Initialize the ScrapeConfigBuilder.
+
+        :param external_url: The external URL to be used for constructing probes' `metrics_path` and `relabel_configs`.
+        """
+        self.external_url = external_url
+
+    def merge_scrape_configs(
+        self, file_probes: Dict[str, Any], relation_probes: List[Dict[str, Any]]
+    ) -> List[Dict[str, Any]]:
+        """Merge the scrape_configs from both file and relation.
+
```

Comment:
Document the order of precedence? Here and in `config.yaml`?

Reply:
added in 8eaaac8170f5fdc10c1d2483b154b262fa145ead